docs/acceptance-criteria/uacs-rfc-19-random-beacon-v2.adoc

```diff
@@ -0,0 +1,467 @@
+
+| Group creation - DKG acceptance
+| When determining validity of the tx unlocking the sortition pool and accepting
+  the DKG result, the following aspects should be taken into account: +
+  - is the sortition pool locked? +
+  - has the challenge period already passed? +
+  Only if all the above conditions have been met, the tx is considered valid.
+
+| Group creation - DKG acceptance
+| If the transaction unlocking the sortition pool and marking the DKG result as
+  accepted gets approved, then: +
+  - the sortition pool should get unlocked +
+  - the DKG result should be accepted and a group should be created based on the
+  candidate group +  
+  - submitter should receive the fixed reward (in T), paid from the maintenance
+  pool to the submitter’s address (in the same tx) +
+// TODO: revisit 2 weeks removal time
+  - operators who became eligible for submitting the DKG result before the
+  member who submitted the DKG result should be punished by being removed from
+  the sortition pool for a fixed amount of time (2 weeks) (in the same tx).
+
+| Group creation - DKG timeout
+| The DKG submission timeout equals the group size multiplied by the number of
+  blocks for a member to become eligible to submit the DKG result. The timer
+  starts at the moment when the first member becomes eligible (the moment when
+  DKG was requested). The timer gets reset when a valid DKG result challenge is
+  submitted. 
+
+| Group creation - DKG timeout
+| Anyone can send a transaction reporting DKG timeout.
+
+| Group creation - DKG timeout
+| Upon noticing the DKG timeout notification, the random beacon contract should
+  determine its validity.
+
+| Group creation - DKG timeout
+| When determining validity of the DKG timeout notification, the following
+  aspects should be taken into account: +
+  - does it reference the existing DKG request? (if not, then notification
+  invalid) +
+  - has the DKG timeout passed? (if not, then notification invalid) +
+  - is the sortition pool in a locked state (if not - meaning somebody already
+  unlocked it - then notification invalid).
+
+| Group creation - DKG timeout
+| If the DKG timeout notification was deemed justified, then: +
+  - the pool should be unlocked +
+  - the fixed amount reward (in T) should be sent from the maintenance pool to
+  the notifier.
+
+| Sortition pool - punishments
+// TODO: revisit 2 weeks removal time
+| Member of the sortition pool can be punished with 2 weeks removal from the
+  pool as a result of: +
+  - misbehavior during the DKG submission +
+  - misbehavior during relay entry submission.
+
+| Sortition pool - punishments
+| Operators punished with removal from the group should not earn rewards for the
+  time of their absence in the pool.
+
+| Sortition pool - punishments
+// TODO: revisit 2 weeks removal time
+| After 2 weeks punish period elapses, operators removed from the sortition pool
+  should be automatically reinstated in the pool and should be able to earn the
```

Comment:
Currently, they need to re-register manually. Is the `automatically reinstated` a hard requirement?

Reply:
No, this is just how I thought it will work, but there were no requirements regarding this, so I believe the manual re-registration is fine. Changed description in 60895c09c5c4dd137f632135b7170b7cd7783f8e.